### PR TITLE
Loop can be terminated after condition is met

### PR DESCRIPTION
**Pull request by larsgrefer**

### CODE REVIEW

File: java/org/apache/catalina/realm/RealmBase.java [hunk 1]
```diff
@@ -756,10 +756,11 @@ public void backgroundProcess() {
                 }
 
                 boolean matched = false;
-                for(int k=0; k < patterns.length && !matched; k++) {
+                for(int k = 0; k < patterns.length; k++) {
```
Comment (michael-o) [inline java/org/apache/catalina/realm/RealmBase.java:759]: Why not make it foreach directly?

File: java/org/apache/catalina/tribes/tipis/AbstractReplicatedMap.java [hunk 2]
```diff
@@ -824,17 +824,23 @@ public void mapMemberAdded(Member member) {
     public boolean inSet(Member m, Member[] set) {
         if ( set == null ) return false;
         boolean result = false;
-        for (int i=0; i<set.length && (!result); i++ )
-            if ( m.equals(set[i]) ) result = true;
+        for (int i = 0; i < set.length; i++ )
```
Comment (michael-o) [inline java/org/apache/catalina/tribes/tipis/AbstractReplicatedMap.java:827]: Same here

File: java/org/apache/catalina/tribes/tipis/AbstractReplicatedMap.java [hunk 3]
```diff
@@ -824,17 +824,23 @@ public void mapMemberAdded(Member member) {
...
         return result;
     }
 
     public Member[] excludeFromSet(Member[] mbrs, Member[] set) {
         List<Member> result = new ArrayList<>();
         for (int i=0; i<set.length; i++ ) {
             boolean include = true;
-            for (int j=0; j<mbrs.length && include; j++ )
-                if ( mbrs[j].equals(set[i]) ) include = false;
+            for (int j = 0; j < mbrs.length; j++ )
```
Comment (michael-o) [inline java/org/apache/catalina/tribes/tipis/AbstractReplicatedMap.java:839]: And here..

File: java/org/apache/catalina/realm/RealmBase.java (hunk 1 above)
Comment (larsgrefer) [reply java/org/apache/catalina/realm/RealmBase.java:759]: I wanted to make a separate PR for all possible foreach loops later

Comment (larsgrefer) [reply java/org/apache/catalina/realm/RealmBase.java:759]: see #273

**Pull request merged**